sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_client.py

```diff
@@ -284,16 +288,21 @@ def unwrap_key(self, algorithm, encrypted_key, **kwargs):
             key = result.key
 
         """
-
-        result = self._client.unwrap_key(
-            vault_base_url=self._key_id.vault_endpoint,
-            key_name=self._key_id.name,
-            key_version=self._key_id.version,
-            algorithm=algorithm,
-            value=encrypted_key,
-            **kwargs
-        )
-        return UnwrapResult(key_id=self.key_id, algorithm=algorithm, key=result.result)
+        local_key = self._get_local_key(**kwargs)
+        if local_key and local_key.is_private_key():
+            if "unwrapKey" not in self._allowed_ops:
+                raise AzureError("This client doesn't have 'keys/unwrapKey' permission")
+            result = local_key.unwrap_key(encrypted_key, **kwargs)
+        else:
+            result = self._client.unwrap_key(
```

Comment:
is `unwrap_key` marked with a @distributed_trace decorator?

Reply:
Yes.

However, I see this PR is somehow removing decorators from other methods. Let me add them back before anyone notices.